src/vertification.ts

```diff
@@ -0,0 +1,274 @@
+                        await member.roles.remove(verifiedRole);
+                        // DM user that they have not paid and thus have been removed
+                        await member.send("You have been unverified from UofGuelph Racing due to not paying the club fee.");
+                        await channel.send(`${member.id} has been unverified.`);
+                    }
+                }
+            }),
+        );
+    });
+}
+
+export async function sendVerificationMessage(member: GuildMember) {
+    const embeds = new EmbedBuilder()
+        .setTitle("UofG Racing Verification")
+        .setDescription("Welcome! To gain access to the server, please verify yourself.")
+        .addFields(
+            { name: "How", value: `1. Apply to this [form](<${FORM_LINK}>).\n2. **DM the bot** the email given to the form.\n3. Follow the instructions given.` },
+            { name: "Accepted emails", value: "**Only @uoguelph.ca** are accepted emails." },
+            { name: "Code expiration", value: "Your code will **expire in 5 minutes**. If it has, please resend your email address and we will send you a new code." },
+        )
+        .setColor("#FFC72A")
+        .setFooter({ text: "UofG Racing will not ask for passwords, credit card information, SSNs, ID, and/or tokens" });
+    return member.send({
+        embeds: [embeds],
+    });
+}
+
+export function generateVerificationCode(user_id: string) {
+    const hash = createHash("sha256");
+    hash.update(user_id + Date.now().toString());
+
+    const chars = "0123456789";
+    const hash_value = hash.digest("hex");
+
+    let code = "";
+    for (let i = 0; i < 7; i++) {
+        const index = parseInt(hash_value.substring(i * 2, i * 2 + 2), 16) % chars.length;
+        code += chars[index];
+    }
+
+    return code;
+}
+
+// Validates the email and make sure it is in the db
+/* eslint-disable no-useless-escape */
+/* eslint-disable no-control-regex */
+export function validateEmail(email: string): boolean {
+    // https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression
+    if (
+        /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/.test(
+            email,
+        )
+    ) {
+        return email.endsWith("@uoguelph.ca");
+    } else {
+        return false;
+    }
+}
+
+// If it is between July-September, allow people who have not paid in. Otherwise,
+// reject :D
+function validateMembership(user_row: Verification): boolean {
+    const current_month = new Date().getMonth();
+
+    if (current_month >= 4 && current_month <= 9) {
```

Comment:
This is May to October (the month count starts at 0)

Reply:
Actively throwing thanks for catching it